Fix L1 fat event filters (90x)

rekovic opened the pull request:
90x version of 81x PR  #16782

thomreis committed:
removing unused simMuonQualityAdjusterDigis module from SimL1TMuon sequence

Conflicts:
	L1Trigger/L1TMuon/data

thomreis committed:
Set correct input collection name for TF muon ntuplizer

Conflicts:
	L1Trigger/L1TMuon/data

thomreis committed:
Add extrapolated muon eta dn phi at the 1st adn 2nd muon station. Extrapolation with https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMuonAnalysisMuonAssociators

Conflicts:
	L1Trigger/L1TMuon/data

thomreis committed:
Update list of triggers to match

Conflicts:
	L1Trigger/L1TMuon/data

rekovic committed:
L1NtupleAOD configuration: add missing configuration Reconstruction_cff needed by l1MuonReco.

Conflicts:
	L1Trigger/L1TMuon/data

thomreis committed:
Add extrapolated phi and eta of muons to L1 ntuples

smuzaffar committed:
remove IN_DICTBUILD with __CLING__ to control the visibility of standard math vs sse

jshlee committed:
ME0Segments changed reference detector from etaPartition to chamber

ME0SegmentAlgorithm.cc - changed ME0EtaPartition -> ME0Chamber, bug fix
for failed fits
ME0SegmentAlgorithmBase.h - changed ME0EtaPartition -> ME0Chamber
ME0SegmentBuilder.cc - changed ME0EtaPartition -> ME0Chamber
ME0SegmentMatcher.cc - changed ME0EtaPartition -> ME0Chamber
ME0SegmentsValidation.cc - changed ME0EtaPartition -> ME0Chamber
muonME0DigisPreReco_cfi.py - etaResolution was set to 0, this gave
me0RecHits 0 for y error, which made ME0Segment fitting fail.

MuonSegFit - increased number of hits that can be fitted

MuonGEMDetLayerGeometryBuilder.cc added chamber
MuonME0DetLayerGeometryBuilder.cc added chamber

GEMSegment.h and ME0Segment.h to use chamberID

fix ME0Geometry.cc layer and chamber functions

clean up GEMSegmentBuilder.cc

MuonTrackLoader.cc added the case for gemSegments

jshlee committed:
fix conflict in ME0PreRecoGaussianModel

rekovic committed:
Add L1TGlobal data files.

rekovic committed:
Add L1T Calorimeter data files.

rekovic committed:
new caloStage2Params 3_3_1 to reflect the change in FW towerEtaMax 28->27 for TowerCount. Set as default. Revert caloStage2Params 3_3 to have as originally towerEtaMax 28 for TowerCount.

jshlee committed:
ME0ReDigiProducer - bug fix position when etaPartition changes
better sanity check and compatable with the oldme0 geometry

bundocka committed:
Fix etSumUnpacker tower count

jshlee committed:
ME0ReDigi changed to realistic settings

ME0ReDigi
added oldYResolution, discretizeY
use flag smearRadial to stear smearing
added smearing for Y after radial smearing
added discretizeY

muonME0ReDigis_cfi.py
set smearTiming to false
set smearRadial to false
set realistic resolutions, newXResolution = 0.03, newYResolution = 1.0

me0RecHits_cfi.py
changed input digi collection to me0ReDigis

bundocka committed:
Fix etSumUnpacker tower count p4

bundocka committed:
Add etem to latest MP unpacker

mandrenguyen committed:
Added 5 TeV pA beamspot and placeholder for 8 TeV

mandrenguyen committed:
Updated pA 8 TeV relval wf, number 281

jshlee committed:
ME0ReDigiProducer - wrong detId was saved

mandrenguyen committed:
Updated 8 TeV vtx smearing parameters

rekovic commented:
please test

cmsbuild commented:
A new Pull Request was created by  @rekovic  for CMSSW_9_0_X.

It involves the following packages:

EventFilter/L1TRawToDigi
HLTrigger/special


@perrotta, @cmsbuild, @silviodonato, @Martin-Grunewald, @rekovic, @fwyzard, @mulhearn, @davidlange6 can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @kreczko this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

rekovic commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

rekovic commented:
+1

fwyzard commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_9_0_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar

cmsbuild commented:
+1
Tested at: f7973e707be34a1f876085f2c0013ee5b1a2ae85
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16799/16661/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16799/16661/summary.html

The workflows 1003.0, 1001.0, 1000.0, 140.53, 136.731, 4.22 have different files in step1_dasquery.log than the ones found in the baseline. You may want to check and retrigger the tests if necessary. You can check it in the "files" directory in the results of the comparisons

mandrenguyen committed:
Mistakenly took 5 TeV numbers before pixel barycenter shift -- fixed

mandrenguyen committed:
sigma z updated

davidlange6 commented on EventFilter/L1TRawToDigi/plugins/L1TValidationEventFilter.cc:
```diff
@@ -102,12 +104,13 @@ L1TValidationEventFilter::filter(edm::Event& iEvent, const edm::EventSetup& iSet
+  evf::evtn::TCDSRecord record((unsigned char *) data);
 
-  bool fatEvent = (header.lvl1ID() % period_ == 0 );
+  bool fatEvent = (record.getHeader().getData().header.triggerCount % period_ == 0 );
```
The unpacking of fed1024 should go into EventFilter/Utilities/plugins/TcdsRawToDigi.cc which can make a data product for you to consume.
@rekovic

davidlange6 replied:
And as an added bonus, the duplicated code below can be removed too

smuzaffar committed:
add missing dictionary std::vector<edm::Ref<std::vector<reco::PFMET> > >

kpedro88 committed:
add option to output CaloSamples as event product for debugging

Commit:
added missing dictionaries which were sliently ignored due to root bug

jshlee committed:
error for y set to 2.5 cm
error at top of chamber set to 300 um
error at bottom of chamber set to ~122 um
make hits stay within etaPartition in y
recover reduction in me0 digis
recover reco muons

kpedro88 committed:
add option to store precise samples for HPDs

kpedro88 committed:
add option to ignore Geant time in HitResponse

jshlee committed:
fixed ME0DigiPreRecoProducer: etaResolution cannot be 0.0 #16838
fixed ME0ReDigiProducer background hit scaling
fixed ME0ReDigiProducer timing

jshlee committed:
fix missing gem hits

jshlee committed:
change time to float

jshlee committed:
update parameters for D1-D5 and D6
increase me0 segment box in eta and time

bundocka committed:
Small fix to MPUnpacker, add HT saturation check in emulator

cmsbuild committed:
Merge pull request #16772 from uhussain/tmp_FixStaticAnaWarns_Nov27

Fixing static analyzer warnings in ElectronEnergyCorrector.cc, GEDPhotonProducer.cc and PhotonMIPHaloTagger.cc

davidlange6 committed:
Merge pull request #16821 from knoepfel/cleanup

Fix clang compiler warning and replace boost::mpl::if_c with std::conditional

Dr15Jones committed:
Fix memory leak in EcalDeadChannelRecoveryNN

The leak was found by valgrind

Dr15Jones committed:
Fix memory leak in EcalRecHitWorkerRecover

Leak found by valgrind

Dr15Jones committed:
Fix memory leak in FastTSGFromPropagation

Leak found by valgrind.

kpedro88 committed:
add option to retrieve HF sampling factor

kpedro88 committed:
add analyzer for CaloSamples

apana committed:
New prescale table with entries for 512 algorithms

apana committed:
New prescale table with entries for 512 algorithms

apana committed:
Switch default to UGT_BASE_RS_PRESCALES_v11.xml

VinInn committed:
Merged ph2_removingPixelPair from repository ebrondol

ebrondol committed:
fix TrajectoryFilter for init and highPt steps

lgray committed:
update MIP timing resolution to 30ps

Commit:
Updated DigitizerUtility::Amplitude object to store the EncodedEventId properly for individual SimHit contributions to the Digi. There was a subtle bug that only the first EncodedEventId was stored. Another bug has also been fixed where the fraction of individual contribution to the digi was calculated wrongly. A few new histograms are added in the validation suite and a few minor bugs corrected

davidlange6 committed:
Merge pull request #16844 from ianna/dd-regression-test-compare

DD Regression Test: Compare Compact Views

davidlange6 committed:
Merge pull request #16860 from ghellwig/fix-misalignment-tool-phase-1_90X

[90X] Fix misalignment tool for phase-1 (and phase-2) configurations.

davidlange6 committed:
Merge pull request #16867 from JanFSchulte/CMSSW_9_0_X

Bring customization of tracking@HLT up to date with 8_1_0

ebrondol committed:
cleaning up

VinInn committed:
Merged ph2_removingPixelPair from repository ebrondol

VinInn committed:
add comments to keep comments

Martin-Grunewald committed:
Workaround for Unrunnable Schedule

cmsbuild committed:
Merge pull request #16862 from iahmad-khan/fix-90x-warnings

removes compiler warnings for unused variables in L1Trigger

cmsbuild committed:
Merge pull request #16852 from Dr15Jones/fixMemoryLeakInL1TGlobalProducer

Fix memory leak in L1TGlobalProducer

cmsbuild committed:
Merge pull request #16883 from lgray/patch-3

Update MIP timing resolution to 30ps

kpedro88 committed:
avoid EDAlias for conditional product

davidlange6 committed:
Merge pull request #16713 from cms-met/MM_VariousFixes_90X_211116

Various small fixes on MET (90X)

davidlange6 committed:
Merge pull request #16865 from iahmad-khan/90x-sizeof-fix

removes sizeof array pointer warning

lgray committed:
move E_cut up to 0.06

boudoul committed:
new XMLs for Phase2 Tracker_4026

boudoul committed:
removing deprec customs and updating names of material in materialanalysis

boudoul committed:
changing from 4021 to 4026

boudoul committed:
rename 4021 to 4026 version

rovere committed:
Extend HitEfficiencyFromHitPattern to PhaseI

The PhaseI detector has more pixel layers compared to the
run2 one: those layers have been added to the efficiency
client via the era mechanism.

rovere committed:
Bug fix

Sam-Harper committed:
making small code clean up fix

boudoul committed:
Updating documentation

smuzaffar committed:
Let cling parse the full TrackingRegion class to avoid errors in root6.08 IBs

include #16890 changes here

boudoul committed:
fix my mistake in material Analyzer

davidlange6 committed:
Merge pull request #16892 from cms-sw/backport-CMSSW_9_0_X-16814

add missing dictionary std::vector<edm::Ref<std::vector<reco::PFMET> > >

davidlange6 committed:
Merge pull request #16893 from cms-sw/backport-CMSSW_9_0_X-16832

added missing dictionaries which were sliently ignored due to root bug

cmsbuild committed:
Merge pull request #16863 from makortel/removePixelTracksForL3Isolation

Remove PixelTracksForL3Isolation configurations and IsolationRegionAroundL3Muon as obsolete

cmsbuild committed:
Merge pull request #16856 from Dr15Jones/fixMemoryLeakOMTFProcessor

Fix per Run memory leak in OMTFProcessor

davidlange6 committed:
Merge pull request #16855 from Dr15Jones/fixMemoryLeakFastSimulationTrackingRecHitProducer

Fix memory leaks in FastSimulation/TrackingRecHitProducer

cmsbuild committed:
Merge pull request #16848 from felicepantaleo/CACell_speedups

Pixel Seeding - CA - small timing improvements

kpedro88 committed:
add option to inject simhits into hcal digitizer

Dr15Jones committed:
Protect TMultiLayerPerceptron from causing a crash at destructor

The TMultiLayerPerceptron creates a TEventList and then deletes it
in its destructor. Unfortunately, the TEventList will attach itself
to the last open TFile and if that file is deleted before the
TMultiLayerPerceptron, the TEventList will be deleted and the
job will crash when the TMultiLayerPerceptron tries to delete it.
This can be avoided by resetting gDirectory to nullptr.

fabozzi committed:
adding 2016H relvals

davidlange6 committed:
Merge pull request #16779 from makortel/configBuilderEarlyDelete

Improve early delete configuration in ConfigBuilder

Dr15Jones committed:
Need to reset gDirectory after constructing TMultiLayerPerceptron

We need to set gDirectory to nullptr before constructing TMultiLayerPerceptron
but other modules calling other ROOT methods need gDirectory to be non-null.

Martin-Grunewald committed:
Remove unneeded include

kpedro88 committed:
add option to inject entire subdet

dsrankin committed:
adding hgcal era modifications

boudoul committed:
put in failsafe to not run PreMixing and HIP simulation simultaneously

davidlange6 committed:
Merge pull request #16902 from fabozzi/from-CMSSW_9_0_X_2016Hrelvals

Adding 2016H relvals [90X]

davidlange6 committed:
Merge pull request #16877 from Dr15Jones/fixMemoryLeakFastTSGFromPropagation

Fix memory leak in FastTSGFromPropagation

davidlange6 committed:
Merge pull request #16853 from Dr15Jones/fixMemoryLeakNuclearInteractionSimulator

fix memoryleak in NuclearInteractionSimulator

cmsbuild committed:
Merge pull request #16840 from lgray/topic_hgcal_clusters

HGCal 2D Layer Clusters

davidlange6 committed:
Merge pull request #16873 from Martin-Grunewald/Fix4ConfDB90X

Fix py files for ConfDB parsing (90X)

davidlange6 committed:
Merge pull request #16777 from Sam-Harper/PackedCandNoPuppi90X

Making Puppi optional for PATPackedCandidate producer : 90X

dsrankin committed:
adding egamma to phase2 list

kpedro88 committed:
restore performance with nonsense value

davidlange6 committed:
Merge pull request #16887 from Martin-Grunewald/90XUnrunnableSchedule

Workaround for Unrunnable Schedule in TSG tests (90X)

Martin-Grunewald committed:
sync with HLT/V758

Martin-Grunewald committed:
fake menus with 8024 template

boudoul committed:
moving 10cm to 1mm in ProdCutsForGamma

lgray committed:
add multiclusters to baseline processing

lgray committed:
simple optimization of HGCalImagingAlgo

lgray committed:
relax e_cut since we have improved the cost of distance function

lgray committed:
remove incorrect include

cmsbuild committed:
Merge pull request #16880 from ebrondol/ph2_fixTrajectoryFilter

Phase 2 - Fix TrajectoryFilter for Initial- and HighPtSteps

davidlange6 committed:
Revert "adding DQM Offline histograms for BTV at HLT"

davidlange6 committed:
Merge pull request #16910 from cms-sw/revert-16636-DQMOfflineBTVatHLT

Revert "adding DQM Offline histograms for BTV at HLT"

smuzaffar committed:
use py2-pippkgs tool to get pip based packges

boudoul committed:
Removing useless xmls files

boudoul committed:
removing useless XMLs from geometry script

lgray committed:
working on integrating HL-LHC beam parameterization from machine folks

Commit:
Updating the DigitizerUtility::Amplitude class + minor change in the validation suite adding a new histogram and changing some binning etc.

kpedro88 committed:
use auto loop

cmsbuild committed:
Merge pull request #16797 from kpedro88/Phase2-WF29

Fix PFTrackAlgoTools memory corruption

cmsbuild committed:
Merge pull request #16906 from boudoul/HIPkiller_90X

put in failsafe to not run PreMixing and HIP simulation simultaneously

gartung committed:
CalibTracker/SiPixelGainCalibration fix clang warning about abs

gartung committed:
Calibration/EcalCalibAlgos fix clang warning about abs

gartung committed:
CaloOnlineTools-EcalTools fix clang warning about abs

gartung committed:
CommonTools-PileupAlgos fix clang warnings about abs and hides overloaded

gartung committed:
DQM-EcalMonitorClient fix clang warnings about abs

gartung committed:
DQM-EcalMonitorTasks fix clang warnings about abs

gartung committed:
DQM-PixelLumi fix clang warnings about abs

gartung committed:
DQMOffline-Ecal fix clang warnings about abs

gartung committed:
EgammaAnalysis-ElectronTools fix clang warnings about abs

gartung committed:
ElectroWeakAnalysis ZEE ZMuMu: fix clang warnings about abs

gartung committed:
misplaced parens

gartung committed:
FastSimulation: fix clang warnings about abs

gartung committed:
Fireworks: fix clang warnings about abs

gartung committed:
GeneratorInterface: fix clang warnings about abs

gartung committed:
Geometry: fix clang warnings about abs

gartung committed:
HLTrigger: fix clang warnings about abs

gartung committed:
MuonAnalysis: fix clang warnings about abs

gartung committed:
PhysicsTools: fix clang warnings about abs

gartung committed:
RecoLocal*: fix clang warnings about abs

gartung committed:
Validation: fix clang warnings about abs

lgray committed:
work in progress

kpedro88 committed:
refactor geometry script to act as its own unit test

kpedro88 committed:
make another loop range-based

mandrenguyen committed:
Update relval_steps.py

smuzaffar committed:
Merge pull request #16912 from cms-sw/smuzaffar-patch-1

use py2-pippkgs tool to get pip based packges

lgray committed:
now reproduces zt distribution of latest HL-LHC beamspot

emiglior committed:
Change of the parameters of SiPixel local reco to match the specs of phase2 ROC

lgray committed:
use M_PI to define 2pi

lgray committed:
make clang happy

lgray committed:
address code review and some refactoring

cmsbuild committed:
Merge pull request #16941 from kpedro88/Phase2-WF31

Reduce CPU time in MuonAssociatorByHitsHelper

cmsbuild committed:
Merge pull request #16930 from gartung/Fireworks-fix-clang-warning-abs

Fireworks: fix clang warnings about abs

cmsbuild committed:
Merge pull request #16919 from gartung/CaloOnlineTools-EcalTools-fix-clang-warning-abs

CaloOnlineTools-EcalTools fix clang warning about abs

cmsbuild committed:
Merge pull request #16918 from gartung/Calibration-EcalCalibAlgos-fix-clang-warning-abs

Calibration/EcalCalibAlgos fix clang warning about abs

cmsbuild committed:
Merge pull request #16917 from gartung/CalibTracker-SiPixelGainCalibration-fix-clang-warning-abs

CalibTracker/SiPixelGainCalibration fix clang warning about abs

cmsbuild committed:
Merge pull request #16896 from boudoul/Phase24026

Updating phase2 Tracker Geometry (T3) - 2023D4 scenario

cmsbuild committed:
Merge pull request #16885 from VinInn/Toward9pre2

PhaseII: optimize seeding, reduce cpu-timing

cmsbuild committed:
Merge pull request #16884 from suchandradutta/Phase2Digitizer_90X_061216

Update of Phase2Digitizer package

cmsbuild committed:
Merge pull request #16876 from Dr15Jones/fixMemoryLeakEcalRecHitWorkerRecover

Fix memory leaks from EcalRecHitProducer

gartung committed:
fix typo

davidlange6 committed:
Merge pull request #16936 from gartung/PhysicsTools-fix-clang-warning-abs

PhysicsTools: fix clang warnings about abs

davidlange6 committed:
Merge pull request #16935 from gartung/MuonAnalysis-fix-clang-warning-abs

MuonAnalysis: fix clang warnings about abs

fabozzi committed:
fixing dataset name

Dr15Jones committed:
Stop gcc 6.2 compiler warnings in PF data products

The gcc 6.2 compiler was complaining about operator<< for various
ParticleFlow data products not being in the proper namespace. As
part of the cleanup of the warnings all the operator<< are no longer
friends of the classes (since they can all use public methods).

gartung committed:
missplaced parens cause abs value of bool

gartung committed:
abs of unsigned uneeded

gartung committed:
added log by mistake

gartung committed:
cast unsigned long to long before abs to prevent undefined behavior

gartung committed:
misplaced parens cause abs of bool

Commit:
Added the differential fC/ADC gain calculation to HCAL QIE8 and QIE11

Dr15Jones committed:
Fix gcc 6.2 compiler warnings for SiPixelObjects

The operator<< needs to be in the same namespace as the class.

Commit:
Avoiding division by 0 (which should be impossible anyway)

gartung committed:
Fix clang warnings about absolute value use.
Replace abs/fabs with std::abs which has a signature for ints and floats.
Fixed misplaced parens which result in absolute values of a bool.
The absolute value of unsigned vars is meaningless. Cast to signed vars
to prevent difference from wrapping to unsigned var max value.

gartung committed:
SimTracker packages: Fix clang warnings about absolute value use.
Replace abs/fabs with std::abs which has a signature for ints and floats.
Fixed misplaced parens which result in absolute values of a bool.
The absolute value of unsigned vars is meaningless. Cast to signed vars
to prevent difference from wrapping to unsigned var max value.

gartung committed:
SimG4Core : Fix clang warnings about absolute value use.
Replace abs/fabs with std::abs which has a signature for ints and floats.
Fixed misplaced parens which result in absolute values of a bool.
The absolute value of unsigned vars is meaningless. Cast to signed vars
to prevent difference from wrapping to unsigned var max value.

gartung committed:
SimMuon packages: Fix clang warnings about absolute value use.
Replace abs/fabs with std::abs which has a signature for ints, doubles and floats.
Fixed misplaced parens which result in absolute values of a bool.
The absolute value of unsigned vars is meaningless. Cast to signed vars
to prevent difference from wrapping to unsigned var max value.

gartung committed:
TrackPropagation packages: Fix clang warnings about absolute value use.
Replace abs/fabs with std::abs which has a signature for ints, doubles and floats.
Fixed misplaced parens which result in absolute values of a bool.
The absolute value of unsigned vars is meaningless. Cast to signed vars
to prevent difference from wrapping to unsigned var max value.

Dr15Jones committed:
Stop gcc 6.2 warnings in PFClusterTools

Made operator<< no longer a friend and declared it in the same
namespace as the class.

Dr15Jones committed:
Stop gcc 6.2 warning in HFShowerLibrary

gcc complained that a statement after an 'if' was not protected
by the 'if'. Moving the statement to a line after stops the warning.

kpedro88 committed:
fix a few bugs

cmsbuild committed:
Merge pull request #16937 from gartung/RecoLocal-fix-clang-warning-abs

RecoLocal*: fix clang warnings about abs

davidlange6 committed:
Merge pull request #16920 from gartung/CommonTools-PileupAlgos-fix-clang-warnings

CommonTools-PileupAlgos fix clang warnings about abs and hides overloaded virtual

davidlange6 committed:
Merge pull request #16891 from cms-sw/backport-CMSSW_9_0_X-16440

remove IN_DICTBUILD with __CLING__ to control the visibility of standard math vs sse

cmsbuild committed:
Merge pull request #16957 from gartung/SimMuon-fix-clang-warning-abs

SimMuon packages: Fix clang warnings about absolute value use.

emiglior committed:
Fix typo in one default value in PixelCPE Generic

mandrenguyen committed:
Update relval_steps.py

Reverting to boosted beamspot with pA 8 TeV vertex smearing.  A separate PR updating the pA autoconditions to use the GT with the corresponding reco BS tag will be submitted momentarily.

jshlee committed:
fix comments
ME0SegmentAlgorithm.cc added cut on dTimeChainBoxMax
muonME0ReDigis_cfi.py instLumi set to 5

jshlee committed:
me0Segments_cfi.py dTimeChainBoxMax set to 15
ME0ReDigiProducer.cc scale only background hits and not smear them

cmsbuild committed:
Merge pull request #16960 from Dr15Jones/fixWarningsSimG4CMSCalo

Stop gcc 6.2 warning in HFShowerLibrary

cmsbuild committed:
Merge pull request #16956 from gartung/SimG4Core-fix-clang-warning-abs

SimG4Core : Fix clang warnings about absolute value use.

cmsbuild committed:
Merge pull request #16954 from gartung/SimTracker-fix-clang-warning-abs

SimTracker packages: Fix clang warnings about absolute value use.

cmsbuild committed:
Merge pull request #16933 from gartung/HLTrigger-fix-clang-warning-abs

HLTrigger: fix clang warnings about abs

cmsbuild committed:
Merge pull request #16931 from gartung/Generator-fix-clang-warning-abs

GeneratorInterface: fix clang warnings about abs

davidlange6 committed:
Merge pull request #16924 from gartung/DQM-PixelLumi-fix-clang-warning-abs

DQM-PixelLumi fix clang warnings about abs

lgray committed:
smarter rehashing in HGCDigitizer

lgray committed:
now includes proper x distribution from HL-LHC folks

lgray committed:
Merge branch 'topic_correct_hllhcbeamspot' of https://github.com/lgray/cmssw into topic_correct_hllhcbeamspot

gartung committed:
change cast to int

davidlange6 committed:
Merge pull request #16950 from gartung/PhysicsTools-fix-clang-warning-abs

PhysicsTools/Utilities/test : misplaced parens cause abs of bool

cmsbuild committed:
Merge pull request #16945 from cms-tsg-storm/90XHLTAfter2016

Update of HLT (PIon) menu (90X)

davidlange6 committed:
Merge pull request #16940 from gartung/Validation-fix-clang-warning-abs

Validation: fix clang warnings about abs

davidlange6 committed:
Merge pull request #16929 from gartung/FastSimulation-fix-clang-warning-abs

FastSimulation: fix clang warnings about abs

davidlange6 committed:
Merge pull request #16928 from gartung/ElectroWeakAnalysis-fix-clang-warning-abs

ElectroWeakAnalysis ZEE ZMuMu: fix clang warnings about abs

davidlange6 committed:
Merge pull request #16922 from gartung/DQM-EcalMonitorClient-fix-clang-warnings-abs

DQM-EcalMonitorClient fix clang warnings about abs

davidlange6 committed:
Merge pull request #16923 from gartung/DQM-EcalMonitorTasks-fix-clang-warnings-abs

DQM-EcalMonitorTasks fix clang warnings about abs

civanch committed:
fix gcc620 warning

gartung committed:
int cast no longer needed since std::abs returns int

kpedro88 committed:
fixes to pass 2023 geom unit test

kpedro88 committed:
replace T1 with T3 (keeping same WF numbers), add deprecation feature in geom script

rekovic committed:
Fix compile error - missing parenthesis.

rekovic committed:
Merged refs/pull/16718/head from repository cms-sw

rekovic committed:
Removing data files.

cmsbuild committed:
Merge pull request #16953 from gartung/SimG4CMS-fix-clang-warning-abs

SimG4CMS packages: Fix clang warnings about absolute value use.

cmsbuild committed:
Merge pull request #16949 from Dr15Jones/fixWarningsPFDataProducts

Stop gcc 6.2 compiler warnings in PF data products

kpedro88 committed:
resolve rebase inconsistency

cmsbuild committed:
Merge pull request #16963 from civanch/fix_gcc620_warning

Fixed gcc620 warning in OscarProducer

davidlange6 committed:
Merge pull request #16948 from fabozzi/CMSSW_9_0_X_fix2016H

Fix input dataset name of DoubleEG 2016H relval

kpedro88 committed:
simplify dependencies

lgray committed:
use average occupancy to determine hashtable sizes

jshlee committed:
fix for background me0 redigi

cmsbuild committed:
Merge pull request #16959 from Dr15Jones/fixWarningsPFClusterTools

Stop gcc 6.2 warnings in PFClusterTools

cmsbuild committed:
Merge pull request #16942 from lgray/topic_correct_hllhcbeamspot

A correct description of the HL-LHC beamspot

cmsbuild committed:
Merge pull request #16932 from gartung/Geometry-fix-clang-warning-abs

Geometry: fix clang warnings about abs

mmusich committed:
Reflect changes in the pA vtx smearing in the 90X GTs

cmsbuild committed:
Merge pull request #16958 from gartung/TrackPropagation-fix-clang-warning-abs

TrackPropagation packages: Fix clang warnings about absolute value use.

cmsbuild committed:
Merge pull request #16951 from igv4321/Proper-hcal-adc-quantization-error

Added the differential fC/ADC gain calculation to HCAL QIE8 and QIE11

davidlange6 committed:
Merge pull request #16927 from gartung/EgammaAnalysis-ElectronTools-fix-clang-warning-abs

EgammaAnalysis-ElectronTools fix clang warnings about abs

cmsbuild committed:
Merge pull request #16815 from VinInn/SpeedP2Clus

Speed up Phase2TrackerClusterizer by a factor 1000

davidlange6 committed:
Merge pull request #16926 from gartung/DQMOffline-Ecal-fix-clang-warnings-abs

DQMOffline-Ecal fix clang warnings about abs

jshlee committed:
me0 tof discrete

jshlee committed:
set smearTiming True

cmsbuild committed:
Merge pull request #16972 from lgray/patch-4

Speed improvement to HGCDigitizer

cmsbuild committed:
Merge pull request #16907 from kpedro88/DebugCaloSamples

HCAL digi simulation debugging tools

davidlange6 committed:
Merge pull request #16965 from kpedro88/Phase2-WF30

Add 2023 geometry unit test, remove T1

gartung committed:
Calibration/EcalAlCaRecoProducers: fix clang warnings
hides overloaded virtual function [-Woverloaded-virtual]
by removing unneeded declarations of functions that do
not override base class functions.

gartung committed:
Calibration/EcalCalibAlgos: potential bug endRun could be called
after correcting parameters to match base class functions
to fix clang warning hides overloaded virtual function [-Woverloaded-virtual]
   virtual void endRun(edm::Run&, const edm::EventSetup&);

davidlange6 committed:
Merge pull request #16974 from cms-AlCaDB/pAVtxSmearingAndAutoCond_90X

PA vtx smearing and auto cond 90x

cmsbuild committed:
Merge pull request #16967 from cms-l1t-offline/pr90x-l1t-integration-v89.15

Pr90x l1t integration (v89.15) fix SumEt, uGT prescales, memory leaks, clean-up

cmsbuild committed:
Merge pull request #16909 from lgray/topic_hgcal_multiclusters

Implement HGCal "Multiclusters"

davidlange6 committed:
Merge pull request #16904 from drankincms/egammadqm_upsg_hgcal

adding hgcal era modifications

davidlange6 committed:
Merge pull request #16605 from jshlee/GEMandME0Segment-bugFix-81x

Gem and me0 segment bug fix 90x

mmusich committed:
update the BeamSpot for the 2023 upgrade simulation with tag BeamSpotObjects_HLLHC_z4p3cm_t193ns_14TeV_nominal_mc

cmsbuild committed:
Merge pull request #17000 from cms-AlCaDB/90X_update2023BS

[90X] update the BeamSpot for the 2023 upgrade simulation

cmsbuild committed:
Merge pull request #16986 from gartung/Calibration-EcalCalibAlgos-fix-clang-warnings

Calibration/EcalCalibAlgos: potential bug endRun could be called

cmsbuild committed:
Merge pull request #16985 from gartung/Calibration-EcalAlCaRecoProducers-fix-clang-warnings

Calibration/EcalAlCaRecoProducers: fix clang warnings

cmsbuild committed:
Merge pull request #16952 from Dr15Jones/fixWarningSiPixelObjects

Fix gcc 6.2 compiler warnings for SiPixelObjects

cmsbuild committed:
Merge pull request #16897 from rovere/FixHitEffHitPatternPhaseI

Fix hit eff hit pattern phase I

davidlange6 committed:
Merge pull request #16943 from emiglior/phase2_pixeldigitzer_parameters_cmssw_90X

Change of the parameters of SiPixel local reco to match the specs of the phase2 read-out chip.

smuzaffar committed:
add the missing dependency for plugin

davidlange6 committed:
Merge pull request #17004 from smuzaffar/from-CMSSW_9_0_X_2016-12-13-1100

add the missing dependency for plugin

rekovic committed:
Merged refs/pull/16799/head from repository cms-sw

rekovic committed:
Now also unpack triggerCount from TCDS.

rekovic committed:
Use unpacked TCDS triggerCount value, as opposed to unpacking TCDS again here.

cmsbuild commented:
Pull request #16799 was updated. @ghellwig, @ianna, @cerminar, @fabozzi, @Martin-Grunewald, @rekovic, @vanbesien, @perrotta, @civanch, @monttj, @silviodonato, @franzoni, @cmsbuild, @fwyzard, @davidlange6, @Dr15Jones, @cvuosalo, @mdhildreth, @slava77, @ggovi, @mmusich, @mulhearn, @arunhep, @dmitrijus, @srimanob, @hengne can you please check and sign again.

rekovic commented:
ooops....will correct.

rekovic committed:
Revert "Use unpacked TCDS triggerCount value, as opposed to unpacking TCDS again here."

This reverts commit f271dafc702ae4b12920368977da8843e5ca7429.

rekovic committed:
Revert "Now also unpack triggerCount from TCDS."

This reverts commit 9f2c6cd9c939835e855fda4d0cd2595553d50151.

rekovic committed:
Revert "Merged refs/pull/16799/head from repository cms-sw"

This reverts commit dcb9fcb37e51126ddd8961624f1d40ea7a7c840e, reversing
changes made to f7973e707be34a1f876085f2c0013ee5b1a2ae85.

cmsbuild commented:
Pull request #16799 was updated. @ghellwig, @lveldere, @ianna, @cerminar, @fabozzi, @Martin-Grunewald, @rekovic, @vanbesien, @govoni, @perrotta, @civanch, @monttj, @silviodonato, @franzoni, @cmsbuild, @fwyzard, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @slava77, @ggovi, @mmusich, @mulhearn, @ssekmen, @arunhep, @perrozzi, @thuer, @dmitrijus, @alja, @srimanob, @hengne can you please check and sign again.

rekovic commented:
will rebase
